src/main/java/skytils/skytilsmod/mixins/MixinEntityRenderer.java

```diff
@@ -47,15 +47,18 @@
         if (Utils.inSkyblock) {
             if (!Utils.inDungeons && Skytils.config.hideCreeperVeilNearNPCs) {
                 List<EntityOtherPlayerMP> npcs = this.mc.theWorld.getPlayers(EntityOtherPlayerMP.class, p -> p.getUniqueID().version() == 2 && p.getHealth() == 20 && !p.isPlayerSleeping());
-                entityList.removeIf(e -> {
-                    if (e instanceof EntityCreeper && e.isInvisible()) {
-                        EntityCreeper entity = (EntityCreeper) e;
-                        if (entity.getMaxHealth() == 20 && entity.getHealth() == 20 && entity.getPowered()) {
-                            return npcs.stream().anyMatch(npc -> npc.getDistanceSqToEntity(entity) <= 7 * 7);
+                for (Entity entity : entityList) {
+                    if (entity instanceof EntityCreeper && entity.isInvisible()) {
+                        final EntityCreeper creeper = (EntityCreeper) entity;
+                        if (creeper.getMaxHealth() == 20 && creeper.getHealth() == 20 && creeper.getPowered()) {
+                            for (EntityOtherPlayerMP npc : npcs) {
+                                if (npc.getDistanceSqToEntity(entity) <= 49) {
+                                    entityList.remove(creeper);
```

Comment:
not too confident on if im removing the correct entity here or not, as i can't really test as this is too late in the game for me lol

Reply:
The feature is not even enabled because of user concerns